R/cfr.r

```diff
@@ -0,0 +1,232 @@
+# Packages ----------------------------------------------------------------
+library(here)
+library(dplyr)
+library(tidyr)
+library(purrr)
+library(vroom)
+library(lubridate)
+library(brms)
+library(parallel)
+library(loo)
+
+# set number of parallel cores
+no_cores <- detectCores()
+
+# Data --------------------------------------------------------------------
+# get raw cases by data of infection from epiforecasts.io 
+case_infs <- vroom(paste0("https://raw.githubusercontent.com/epiforecasts/covid-rt-estimates/",
+                          "master/subnational/united-kingdom-local/cases/summary/cases_by_infection.csv")) %>% 
+  mutate(data = "cases")
+# get raw deaths by data of infection from epiforecasts.io 
+death_infs <- vroom(paste0("https://raw.githubusercontent.com/epiforecasts/covid-rt-estimates/",
+                           "master/subnational/united-kingdom-local/deaths/summary/cases_by_infection.csv")) %>% 
+  mutate(data = "deaths")
+
+# link datasets and pivot wider
+infections <- case_infs %>% 
+  bind_rows(death_infs) %>% 
+  filter(type == "estimate") %>% 
+  select(utla_name = region, date, data, value = median) %>% 
+  pivot_wider(names_from = "data") %>% 
+  group_by(utla_name) %>% 
+  complete(date = seq(min(date), max(date), by = "day")) %>% 
+  mutate(cases = replace_na(cases, 0)) %>% 
+  drop_na(deaths)
+
+# Define covariates -------------------------------------------------------
+# get variant proportion
+sgene_by_utla <- readRDS(here("data", "sgene_by_utla.rds")) %>% 
+  drop_na(prop_sgtf) %>% 
+  filter(week_infection > "2020-10-01")
+
+# make infections weekly summary
+weekly_infections <- infections %>% 
+  mutate(week_infection = floor_date(date, "week", week_start = wday(max(sgene_by_utla$week_infection)) - 1)) %>% 
+  group_by(utla_name, week_infection) %>%
+  summarise(cases = sum(cases), deaths = sum(deaths), n = n(), .groups = "drop") %>%
+  filter(n == 7) %>%
+  select(-n)
+
+# link with variant proportion
+deaths_with_cov <- weekly_infections %>% 
+  inner_join(sgene_by_utla, by = c("week_infection", "utla_name")) %>% 
+  select(utla = utla_name, week_infection, region = nhser_name, 
+         deaths, cases, prop_sgtf, samples)
+
+# make normalised predictors
+deaths_with_cov <- deaths_with_cov %>% 
+  mutate(normalised_cases = (cases - mean(cases)) / sd(cases),
+         time = as.numeric(week_infection),
+         time = time - min(time),
+         time = (time - mean(time)) / sd(time))
+
+# Define model ------------------------------------------------------------
+# D = c^{+}(1-f)C + c^{-}fC
+# where c is the CFR, f is the fraction of cases that are sgtf, C is the number
+# of cases by date of infection, and D is the number of deaths by date of infection
+# effect of the variant on the CFR is assumed to be:
+# c^{-} = \alpha c^{+} or c^{-} = \alpha + c^{+}
+# leading to:
+# D = (1 - (\alpha - 1)f)c^{+}C or D = (c^{+} + \alpha f)C 
```

Comment:
should be D = (1 + (\alpha - 1)f)c^{+}C
(i.e., I think it's correct in the function below)